### PR TITLE
Finish RPC support in raw sensor log

### PR DESCRIPTION
**Pull request by timburke**
### Overview

This PR finishes the remaining RPC support needed for a complete sensor log controller subsystem.  It also adds the required features and tests into `iotile-sensorgraph` to implement this feature.  It also adds support for `fill-stop` mode and finishes the config variable streaming support to properly latch config variables as python objects on tile reset.

Keys Changes:

- Adds remaining RPCs into raw sensor log (`highest_id` and `seek_stream`).
- Implementing these RPCs required adding features to BufferedStreamWalker and InMemoryStorageEngine in `iotile-sensorgraph` to support the required search and seek behavior with appropriate exceptions for the conditions that we need to flag out of the RPC.
- Increases test coverage of `iotile-sensorgraph`
- Adds test coverage of new RPCs.
- Finishes support for config variable streaming on reset.  Adds a set of new functionality to convert binary config variables into python objects using `latch_config_variables()`.
- Adds significant test coverage over `latch_config_variables` and related functionality
- Starts adding a list of global stream name constants similar to how global RPCs are defined.   
- Adds a `stream_name` function to generate a nice string with the common name for a given stream.

Addresses most of #560 (remaining part is `dump/restore` behavior for state serialization).

### CODE REVIEW

File: iotileemulate/iotile/emulate/constants/streams.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    """Create a StreamDeclaration from a string name.
+
+    This will encode the string name into a 16-bit stream
+    identifier.
+
+    Args:
+        string_name (str): The human-readable name of the stream.
+
+    Returns:
+        StreamDeclaration: The stream declaration.
```
Comment (timburke) [inline iotileemulate/iotile/emulate/constants/streams.py]:
```suggestion
        int: The stream declaration.
```

File: iotileemulate/test/test_constants.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from iotile.emulate.constants import stream_name, rpc_name
+
+
+def test_stream_name_foramatting():
```
Comment (timburke) [inline iotileemulate/test/test_constants.py]:
```suggestion
def test_stream_name_formatting():
```

File: iotilesensorgraph/iotile/sg/sensor_log.py [hunk 3]
```diff
@@ -18,6 +18,15 @@
 class SensorLog(object):
     """A storage engine holding multiple named FIFOs.
 
+    Normally a SensorLog is used in ring-bufffer mode which means that old
```
Comment (mattrunchey) [inline iotilesensorgraph/iotile/sg/sensor_log.py]: typo - "buffer"

File: iotilesensorgraph/iotile/sg/sensor_log.py [hunk 4]
```diff
@@ -46,6 +55,32 @@ def __init__(self, engine=None, model=None):
 
         self._engine = engine
         self._model = model
+        self._rollover_streaming = True
+        self._rollover_storage = True
+
+    def set_rollover(self, area, enabled):
+        """Configure whether rollover is enabled for streaming or storage streams.
+
+        Normally a SensorLog is used in ring-bufffer mode which means that old
```
Comment (mattrunchey) [inline iotilesensorgraph/iotile/sg/sensor_log.py]: same - bufffer = buffer

**Comment by timburke** [comment]
> Quick / general question -- is the README.md meant to be blank?

Good point.  The readme should be updated to have a high level intro as you say.  I've just been waiting until I stabilize the core feature set first and then will document in the readme before releasing.  These PRs are all just getting things into master in preparation for a v0.1.0 release of `ReferenceDevice` once it supports all necessary controller subsystems so you could really use it.

File: iotilesensorgraph/iotile/sg/sensor_log.py (hunk 3 above)
Comment (timburke) [inline iotilesensorgraph/iotile/sg/sensor_log.py]:
```suggestion
    Normally a SensorLog is used in ring-buffer mode which means that old
```

Comment (timburke) [inline iotilesensorgraph/iotile/sg/sensor_log.py]:
```suggestion
    Normally a SensorLog is used in ring-buffer mode which means that old
```

File: iotilesensorgraph/iotile/sg/sensor_log.py [hunk 5]
```diff
@@ -46,6 +57,104 @@ def __init__(self, engine=None, model=None):
...
+        for walker in self._queue_walkers:
+            if walker.selector in old_walkers:
+                walker.restore(old_walkers[walker.selector])
+            elif not permissive:
+                raise ArgumentError("Cannot restore SensorLog, walker %s exists in restored log but did not exist before" % str(walker.selector))
+
+    def set_rollover(self, area, enabled):
+        """Configure whether rollover is enabled for streaming or storage streams.
+
+        Normally a SensorLog is used in ring-bufffer mode which means that old
```
Comment (timburke) [inline iotilesensorgraph/iotile/sg/sensor_log.py]:
```suggestion
        Normally a SensorLog is used in ring-buffer mode which means that old
```

File: iotilesensorgraph/iotile/sg/sensor_log.py (hunk 4 above)
Comment (timburke) [reply iotilesensorgraph/iotile/sg/sensor_log.py]: Thanks for finding these.  I just fixed both of them.  One suggestion could be to use the github `suggestion` to just put the fix inline and then I can just click accept.  I've started doing that on my own PRs as I read through the diffs to quickly fix typos

image

Comment (mattrunchey) [reply iotilesensorgraph/iotile/sg/sensor_log.py]: Oh awesome -- will definitely use that!

**Pull request merged**